api/agent/slots.go

```diff
@@ -352,6 +354,11 @@ func getSlotQueueKey(call *call) string {
 		hash.Write(unsafeBytes("\x00"))
 	}
 
+	if driver != nil {
+		slotExtns := driver.GetSlotKeyExtensions(call.Extensions())
```

Comment:
Some part of me finds it peculiar to pass in `driver` just to invoke `GetSlotKeyExtensions()` when we have both `driver` and `call` present in the callsite at `agent.getSlot()`.  Would it be cleaner to hoist this out into `getSlot()` and then just pass the filtered string into `getSlotQueueKey()`?  Or are we not guaranteed to have all of the pieces together in all cases?